qiskit/compiler/transpiler.py

```diff
@@ -856,14 +861,27 @@ def _parse_output_name(output_name, circuits):
         return [circuit.name for circuit in circuits]
 
 
-def _parse_alignment(backend, alignment, num_circuits):
-    if backend is None and alignment is None:
-        alignment = 1
+def _parse_pulse_constraints(backend, pulse_constraints, num_circuits):
+
+    if backend is None and pulse_constraints is None:
+        pulse_constraints = PulseConstraints()
     else:
-        if alignment is None:
-            alignment = getattr(backend.configuration(), "alignment", 1)
+        if pulse_constraints is None:
+            # get constraints from backend
+            pulse_constraints = getattr(
+                backend.configuration(),
+                "pulse_constraints",
+                PulseConstraints(),
+            )
+    pulse_constraints = PulseConstraints(**pulse_constraints)
 
-    if not isinstance(alignment, int) or alignment == 0:
-        raise TranspilerError(f"Alignment should be nonzero integer value. Not {alignment}.")
+    # validation
+    fields = ["granularity", "min_length", "control_alignment", "measure_alignment"]
+    for field in fields:
+        constraint_value = getattr(pulse_constraints, field, None)
+        if not isinstance(constraint_value, int) or constraint_value < 1:
+            raise TranspilerError(
+                f"Pulse constraint {field} should be nonzero integer. Not {constraint_value}."
+            )
```

Comment:
I think this validation should go inside PulseConstraints itself during construction. Also make it fast when there's no constraint.

Reply:
Thanks, this is moved to the constructor https://github.com/Qiskit/qiskit-terra/pull/6705/commits/0c38fcd0dac96e900713b5301cf8b791f495f86f